streams/src/test/java/org/apache/kafka/streams/integration/TaskAssignorIntegrationTest.java

```diff
@@ -142,4 +169,184 @@ public void shouldProperlyConfigureTheAssignor() throws NoSuchFieldException, Il
             assertThat(taskAssignor, instanceOf(MyTaskAssignor.class));
         }
     }
+
+    @Test
+    public void shouldScaleOutWithWarmupTasksAndInMemoryStores() throws InterruptedException {
```

Comment:
Here are the new tests, one for in-memory and one for persistent. We expect exactly the same behavior for both.